common/cookie.re

```diff
@@ -0,0 +1,86 @@
+type document;
+
+/*
+ This is used for establishing a well-typed interface for
+ the content of a cookie, which needs to be in a specific
+ format. We express any cookie content as type Content.t
+ instead of a string type.
+
+ Advantage of this approach is that we don't need to worry
+ about wrongly passing strings. Everytime we want to write
+ a cookie, we need to use the Content module first.
+ */
+module type ContentSig = {
+  type t;
+  let make: (~name: string, ~value: string, ~days: int) => t;
+  let fromString: string => t;
+  let read: (t, string) => option(string);
+  let toString: t => string;
+};
+
+module Content: ContentSig = {
+  exception StopRead(string);
+
+  type t = string;
+
+  /* See https://www.quirksmode.org/js/cookies.html */
+  let make = (~name: string, ~value: string, ~days: int) => {
+    let expires =
+      if (days > 0) {
+        open Js.Date;
+        let futureTime =
+          make()->getTime
+          +. float_of_int(days * 24 * 60 * 60 * 1000)
+          |> fromFloat;
+
+        "; expires=" ++ futureTime->toUTCString;
+      } else {
+        "";
+      };
+    {j|$name=$value$expires; path=/|j};
+  };
+
+  let read = (content, name) => {
+    /* Note: This is a hacky approach of doing imperative programming.
+       I was literally translating a very imperative example from JS
+       to Reason. There are ways to rewrite it in a functional way which
+       may be much cleaner */
+
+    open Js.String;
+    let nameEQ = name ++ "=";
+    let ca = split(";", content);
```

Comment:
```suggestion
    let ca = split("; ", content);
```

Reply:
Not sure why this is important? we trim all tokens in line 55 anyways?

Reply:
sorry. did not see that. cleaner too.  :)